build-system/tasks/process-github-issues.js

```diff
@@ -0,0 +1,274 @@
+            issueType = label.name;
+          }
+          // Check if the issues has Priority
+          if (label.name.startsWith('P0') || label.name.startsWith('P1')
+            || label.name.startsWith('P2') || label.name.startsWith('P3')) {
+            hasPriority = true;
+          }
+        }
+      });
+      promise = promise.then(function() {
+        util.log('Update ' + issue.number);
+        var updates = [];
+        // Milestone task: move issue from closed milestone
+        if (milestone) {
+          if (milestoneTitle.startsWith('Sprint') &&
+            milestoneState === 'closed') {
+            // 4 is the number for Milestone 'Backlog Bugs'
+            issueNewMilestone = 4;
+            updates.push(applyMilestone(issue, issueNewMilestone));
+          }
+        }
+        //if issueType is not null, add correct milestones
+        if (issueType != null) {
+          if (milestoneTitle === 'Pending Triage' || milestone == null) {
+            if (issueType === 'Type: Feature Request') {
+              // 23 is the number for Milestone 'New FRs'
+              issueNewMilestone = 23;
+              updates.push(applyMilestone(issue, issueNewMilestone));
+            }
+            else if (issueType === 'Related to: Documentation' ||
+              issueType === 'Type: Design Review' ||
+              issueType === 'Type: Weekly Status') {
+              // 12 is the number for Milestone 'Docs Updates'
+              issueNewMilestone = 12;
+              updates.push(applyMilestone(issue, issueNewMilestone));
+            }
+            else if (issueType === 'Type: Bug' ||
+              issueType === 'Related to: Flaky Tests') {
+              // 4 is the number for Milestone 'Backlog Bugs'
+              issueNewMilestone = 4;
+              updates.push(applyMilestone(issue, issueNewMilestone));
+            }
+            else if (milestone == null) {
+              // 20 is the number for Milestone 'Pending Triage'
+              updates.push(applyMilestone(issue, issueNewMilestone));
+            }
+          }
+        }
+        else if (milestone == null) {
+          // 20 is the number for Milestone 'Pending Triage'
+          updates.push(applyMilestone(issue, issueNewMilestone));
+        }
+        else if (milestoneTitle === 'Prioritized FRs' ||
+          milestoneTitle === 'New FRs') {
+          updates.push(applyLabel(issue, 'Type: Feature Request'));
+        }
+        else if (milestoneTitle === 'Backlog Bugs' ||
+          milestoneTitle.startsWith('Sprint')) {
+          updates.push(applyLabel(issue, 'Type: Bug'));
+        }
+        // Apply default priority if no priority
+        if (hasPriority == false && milestoneTitle != 'New FRs' &&
+          milestoneTitle !== '3P Implementation' &&
+          milestoneTitle !== 'Pending Triage' && milestone != null) {
+          updates.push(applyLabel(issue, 'P2: Soon'));
+        }
+        return Promise.all(updates);
```

Comment:
Maybe add the dry run code here? (If it's a dry run, you'd want to print out what's going to happen without actually doing it.)

Reply:
In the dryrun case, you can simply print every `request` object in `updates` and return an empty promise, without actually executing the requests. Something like the following should work.
```
if (isDryrun) {
  util.log('Performing a dry run. These are the updates that would have been applied:');
  updates.forEach(function(update) {
    util.log(util.inspect(update, { depth: null }));
  });  
  return Promise.resolve();
} else {
  return Promise.all(updates);
}
```